src/core/createConfig.js

```diff
@@ -57,9 +57,31 @@ const createConfig = config => {
     /**
      * Validates the configuration against the defined schema.
      */
-    validate: () => {
-      // TODO: Validate existing configuration against defined schema.
-      return true;
+    validate: (schemaObj, key) => {
+      if (!schemaObj) {
+        cfg.validate(cfg.schema);
+        return;
+      }
+      const currentKey = key || "";
+      const keys = Object.keys(schemaObj);
+      const required = schemaObj.R;
+      const defaultValue = schemaObj.D;
+      const currentValue = cfg.get(currentKey);
+      if (!currentValue && currentKey) {
```

Comment:
Right, I noticed that after the commit. I didn't think about null though. I'm not sure if that would be specified unless introduced during runtime.

Reply:
I checked json-schema and `undefined` is also considered a valid value when a property is marked `required`. If we follow json-schema, we would use `hasOwnProperty` to see if the property is on the object as all. If so, then it would comply with the `required` requirement.